CMakeLists.txt

```diff
@@ -84,7 +84,7 @@ target_link_directories(onnxruntime-genai PRIVATE ${ORT_LIB_DIR})
 
 # we keep the shared libraries disconnected on Android as they will come from separate AARs and we don't want to force
 # the ORT version to match in both.
-if(NOT ANDROID)
+if(NOT (CMAKE_SYSTEM_NAME STREQUAL "Android" OR CMAKE_SYSTEM_NAME STREQUAL "Linux"))
```

Comment:
Why not do this for windows also? Then we have only one solution instead of 2.

Reply:
Mostly because `dlopen` isn't available on Windowsl. We'll need to use `LoadLibraryA` or something like that. You're right, one solution does sounds better.

Reply:
Let's do it in another PR. I think this PR is good to go. But we can wait for @snnn to share his thoughts as well.

Reply:
Technically this PR lacks more fine-grained searching for libraries. In #662 you added the search path like this for wheel scenario:

```cpp
 set_property(TARGET python APPEND_STRING PROPERTY LINK_FLAGS " -Xlinker -rpath=\\$ORIGIN/../onnxruntime/capi/")
```

With this PR, we'll need to search for `../onnxruntime/capi` in C++. While waiting for Changming, I'll add the necessary searching accroding to #662 today.